packages/vm/src/evm/opcodes/util.ts

```diff
@@ -261,3 +262,70 @@ export function updateSstoreGas(
     return new BN(common.param('gasPrices', 'sstoreSet'))
   }
 }
+
+export const eof1CodeAnalysis = (container: Buffer) => {
+  const magic = 0x00
+  const version = 0x01
+  const secCode = 0x01
+  const secData = 0x02
+  const secTerminator = 0x00
+  let computedContainerSize = 0
+  const sectionSizes = {
+    code: 0,
+    data: 0,
+  }
+  if (container[1] === magic && container[2] === version) {
+    if (container.length > 7 && container[3] === secCode && container[6] === secTerminator) {
+      sectionSizes.code = (container[4] << 8) | container[5]
+      computedContainerSize = 7 + sectionSizes.code
+      // Code size cannot be 0
+      if (sectionSizes.code < 1) return
+    } else if (
+      container.length > 10 &&
+      container[3] === secCode &&
+      container[6] === secData &&
+      container[9] === secTerminator
+    ) {
+      sectionSizes.code = (container[4] << 8) | container[5]
+      sectionSizes.data = (container[7] << 8) | container[8]
+      computedContainerSize = 10 + sectionSizes.code + sectionSizes.data
+      // Code & Data sizes cannot be 0
+      if (sectionSizes.code < 1 || sectionSizes.data < 1) return
+    }
+    if (container.length !== computedContainerSize) {
+      // Scanned code does not match length of contract byte code
+      return
+    }
+    return sectionSizes
+  }
+}
```

Comment:
This is checking on so many conditions in such a condensed form that it is extremely hard to review and generally follow. Can we have this dramatically [TM] more commented, so basically to a comment with a condition on more or less every line of code? Guess that would make things easier to grasp.

Also: one technique @ryanio often applies and I guess which would help here as well is to instead of checking for the positive condition (`if (container[1] === magic && container[2] === version) {`) and then nesting further down check for the negative (`if (container[1] !== magic || container[2] !== version) {`) and the directly do the `return`. This prevents further nesting and could be minimally applied on this example I made but I guess also on other code parts.